types/src/key.rs

```diff
@@ -899,13 +899,14 @@ impl Key {
             Err(error) => return Err(FromStrError::EntryPoint(error.to_string())),
         }
 
-        if let Some(contract_entity_hash) = input.strip_prefix(STATE_PREFIX) {
-            let package_addr_bytes = checksummed_hex::decode(contract_entity_hash)
-                .map_err(|error| FromStrError::State(error.to_string()))?;
-
-            let entity_hash_addr: HashAddr = PackageAddr::try_from(package_addr_bytes.as_ref())
-                .map_err(|error| FromStrError::Package(error.to_string()))?;
-            return Ok(Key::State(EntityAddr::SmartContract(entity_hash_addr)));
+        if let Some(entity_addr_formatted) = input.strip_prefix(STATE_PREFIX) {
```

Comment:
I piggy backed this change because it makes sidecars tests fail.
I consulted with @mpapierski that deserialising always to Key::State(EntityAddr::SmartContract) is a bug, so I changed it that it will consider all EntityAddr possibilities.